### ISSUE TITLE
Error on Linux build with 6.0.6+1 

### ISSUE THREAD

**Issue by jnet06** [singerdmx/flutter-quill#967]
Hello, 
My issue is about publishing with Linux using snapcraft.

I have tried running snapcraft with flutter_quill 6.0.6+1 but I keep getting this error:

Error: The non-abstract class 'RawEditorState' is missing implementations for these members:
[   +1 ms] [        ]  - TextInputClient.performSelector
[        ] [        ] Try to either
[        ] [        ]  - provide an implementation,
[        ] [        ]  - inherit an implementation from a superclass or mixin,
[        ] [        ]  - mark the class as abstract, or
[        ] [        ]  - provide a 'noSuchMethod' implementation.
[        ] [        ] class RawEditorState extends EditorState
[        ] [        ]       ^^^^^^^^^^^^^^
[        ] [        ] ../../../snap/flutter/common/flutter/packages/flutter/lib/src/services/text_input.dart:1172:8: Context: 'TextInputClient.performSelector' is defined here.
[        ] [        ]   void performSelector(String selectorName) {}
[        ] [        ]        ^^^^^^^^^^^^^^^

Would appreciate some help.
Thanks!

**Comment by piesuke** [comment]
I have a same problem.
When `flutter run`, I got these errors.

```
../../../../../../.pub-cache/hosted/pub.dartlang.org/flutter_quill-
6.0.6+1/lib/src/widgets/raw_editor.dart:238:7: Error: The
non-abstract class 'RawEditorState' is missing implementations for
these members:
 - TextInputClient.performSelector
Try to either
 - provide an implementation,
 - inherit an implementation from a superclass or mixin,
 - mark the class as abstract, or
 - provide a 'noSuchMethod' implementation.

class RawEditorState extends EditorState
      ^^^^^^^^^^^^^^
../../../../../../development/flutter/packages/flutter/lib/src/serv
ices/text_input.dart:1172:8: Context:
'TextInputClient.performSelector' is defined here.
  void performSelector(String selectorName) {}
       ^^^^^^^^^^^^^^^
../../../../../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib
/src/interface/file.dart:15:16: Error: The method 'File.create' has
fewer named arguments than those of overridden method
'File.create'.
  Future<File> create({bool recursive = false});
               ^
org-dartlang-sdk:///dart-sdk/lib/io/file.dart: Context: This is the
overridden method ('create').
../../../../../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib
/src/interface/file.dart:15:16: Error: The method 'File.create'
doesn't have the named parameter 'exclusive' of overridden method
'File.create'.
  Future<File> create({bool recursive = false});
               ^
org-dartlang-sdk:///dart-sdk/lib/io/file.dart: Context: This is the
overridden method ('create').
../../../../../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib
/src/forwarding/forwarding_file.dart:22:8: Error: The method
'ForwardingFile.createSync' has fewer named arguments than those of
overridden method 'File.createSync'.
  void createSync({bool recursive = false}) =>
       ^
org-dartlang-sdk:///dart-sdk/lib/io/file.dart: Context: This is the
overridden method ('createSync').
../../../../../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib
/src/forwarding/forwarding_file.dart:22:8: Error: The method
'ForwardingFile.createSync' doesn't have the named parameter
'exclusive' of overridden method 'File.createSync'.
  void createSync({bool recursive = false}) =>
       ^
org-dartlang-sdk:///dart-sdk/lib/io/file.dart: Context: This is the
overridden method ('createSync').
../../../../../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib
/src/backends/memory/memory_file.dart:57:8: Error: The method
'MemoryFile.createSync' has fewer named arguments than those of
overridden method 'File.createSync'.
  void createSync({bool recursive = false}) {
       ^
org-dartlang-sdk:///dart-sdk/lib/io/file.dart: Context: This is the
overridden method ('createSync').
../../../../../../.pub-cache/hosted/pub.dartlang.org/file-6.1.2/lib
/src/backends/memory/memory_file.dart:57:8: Error: The method
'MemoryFile.createSync' doesn't have the named parameter
'exclusive' of overridden method 'File.createSync'.
  void createSync({bool recursive = false}) {
       ^
org-dartlang-sdk:///dart-sdk/lib/io/file.dart: Context: This is the
overridden method ('createSync').
```

My flutter and dart version are

```
Flutter 3.4.0-17.2.pre • channel beta •
ssh://git@github.com/flutter/flutter.git
Framework • revision d6260f127f (12 days ago) • 2022-09-21 13:33:49
-0500
Engine • revision 3950c6140a
Tools • Dart 2.19.0 (build 2.19.0-146.2.beta) • DevTools 2.16.0
```

**Comment by Namli1** [comment]
Facing the same issue on iOS, with flutter_quill version 6.1.6 and Flutter version 3.5.

```
Failed to build iOS app
Error (Xcode): ../../../../.pub-cache/hosted/pub.dev/flutter_quill-6.1.6/lib/src/widgets/raw_editor.dart:240:7: Error: The non-abstract class 'RawEditorState' is missing implementations for these members:

Could not build the application for the simulator.
Error launching application on iPhone 14 Pro.
```
I can't really say why this suddenly happened, but it only appeared once I upgraded to Flutter 3.

Would love for this to be resolved and thanks for all the work, flutter_quill is really the best WYSIWYG editor for Flutter!

**Comment by Namli1** [comment]
Okay, I just looked into it and the problem was that I wasn't on the flutter stable branch. 
With the versions below it works for me on iOS:
```
flutter_quill: ^6.1.6

Flutter 3.3.9 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b8f7f1f986 (2 weeks ago) • 2022-11-23 06:43:51 +0900
Engine • revision 8f2221fbef
Tools • Dart 2.18.5 • DevTools 2.15.0
```
You should probably test if this works for you too on Linux. 

The problem is that on the master branch of Flutter, there are some new methods of `TextInputClient` that require implementation, so the app won't build.

**Comment by Namli1** [comment]
Okay, I just looked into it and the problem was that I wasn't on the flutter stable branch. 
With the versions below it works for me on iOS:
```
flutter_quill: ^6.1.6

Flutter 3.3.9 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b8f7f1f986 (2 weeks ago) • 2022-11-23 06:43:51 +0900
Engine • revision 8f2221fbef
Tools • Dart 2.18.5 • DevTools 2.15.0
```
You should probably test if this works for you too on Linux. 

The problem is that on the master branch of Flutter, there are some new methods of `TextInputClient` that require implementation, so the app won't build.

**Comment by aitaro** [comment]
This issue seems to be resolved by #1041.
